File: src/ObjectManager/Object.Tes/FilePacks/Records/005-DLBR.Dialog Branch.cs
using OA.Core;
using System;

namespace OA.Tes.FilePacks.Records
{
    public class DLBRRecord : Record
    {
        public override string ToString() => $"DLBR: {EDID.Value}";
        public STRVField EDID; // Action Name
        public CREFField CNAME; // RGB color

        public override bool CreateField(UnityBinaryReader r, string type, uint dataSize) => throw new NotImplementedException();
        public override bool CreateField(UnityBinaryReader r, GameFormatId formatId, string type, uint dataSize)
        {
            switch (type)
            {
                case "EDID": EDID = new STRVField(r, dataSize); return true;
                case "CNAME": CNAME = new CREFField(r, dataSize); return true;
                default: return false;
            }
        }
    }
}

File: src.mac/ObjectManager/ObjectManager/Object.Tes/FilePacks/Records/005-DLBR.Dialog Branch.swift
//
//  DLBRRecord.swift
//  ObjectManager
//
//  Created by Sky Morey on 5/28/18.
//  Copyright © 2018 Sky Morey. All rights reserved.
//

public class DLBRRecord: Record {
    public override var description: String { return "DLBR: \(EDID)" }
    public var EDID: STRVField = STRVField.empty // Editor ID
    public var CNAME: CREFField! // RGB color
    
    override func createField(_ r: BinaryReader, for format: GameFormatId, type: String, dataSize: Int) -> Bool {
        switch type {
        case "EDID": EDID = STRVField(r, dataSize)
        case "CNAME": CNAME = r.readT(dataSize)
        default: return false
        }
        return true
    }
}
